### PR TITLE
[DO-1702]Storing basic data for Declaration model

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        return None
+
+    def save_or_update_declaration(self):
+        for nacp_declarant_id in self.only_peps:
+            # getting general info including declaration id
+            response = requests.get(
+                settings.NACP_DECLARATION_LIST
+                + f'?user_declarant_id={nacp_declarant_id}'
+            )
+            if (response.status_code != 200):
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:48]: maybe `response.status_code == 404`?? more clear for message `f'cannot find declarations of the PEP with nacp_declarant_id`

File: business_register/converter/declaration.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                logger.error(
+                    f'cannot find declarations of the PEP with nacp_declarant_id: {nacp_declarant_id}'
+                )
+                continue
+            for declaration_data in response.json()['data']:
+                declaration_id = declaration_data['id']
+                declaration = self.all_declarations.get(declaration_id)
+                # ToDo: predict storing changes from the declarant
+                if declaration:
+                    continue
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:58]:
```suggestion
                # ToDo: predict storing changes from the declarant
                if declaration_id in self.all_declarations:
                    continue
```

File: business_register/converter/declaration.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    year=declaration_data['declaration_year'],
+                    nacp_declaration_id=declaration_id,
+                    nacp_declarant_id=nacp_declarant_id,
+                    pep=self.only_peps[nacp_declarant_id],
+                    last_job_title=declaration_data['data']['step_1']['data']['workPost'],
+                    last_employer=declaration_data['data']['step_1']['data']['workPlace'],
+                )
+                # getting full declaration data
+                response = requests.get(settings.NACP_DECLARATION_RETRIEVE + declaration_id)
+                if (response.status_code != 200):
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:70]:
```suggestion
                if response.status_code != 200:
```
```suggestion
                if response.status_code == 404:
```

File: business_register/models/declaration_models.py [hunk 4]
```diff
@@ -37,7 +37,6 @@ class Declaration(DataOceanModel):
     )
     nacp_declarant_id = models.PositiveIntegerField(
         _('NACP id of the declarant'),
-        unique=True,
         db_index=True
     )
```
Comment (roman-tiukh) [inline business_register/models/declaration_models.py:41]: why not unique?

File: business_register/management/commands/fetch_peps_nacp_id.py [hunk 5]
```diff
@@ -57,9 +56,9 @@ def handle(self, *args, **options):
                 if pep.nacp_id:
                     continue
                 declaration_id = pep_data[1].replace('nacp_', '')
-                response = requests.get(self.NACP_URL + declaration_id)
+                response = requests.get(settings.NACP_DECLARATION_RETRIEVE + declaration_id)
                 if (response.status_code != 200):
```
Comment (roman-tiukh) [inline business_register/management/commands/fetch_peps_nacp_id.py:60]:
```suggestion
                if response.status_code != 200:
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/declaration.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DeclarationConverter(BusinessConverter):
+
+    def __init__(self):
+        self.only_peps = {
+            getattr(pep, 'source_id'): pep for pep in Pep.objects.filter(is_pep=True)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:17]: thank you), changed

File: business_register/converter/declaration.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        }
+        self.all_declarations = self.put_objects_to_dict(
+            'nacp_declaration_id',
+            'business_register',
+            'Declaration'
+        )
+
+    # ToDo: extract registration data
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:25]: changed

File: business_register/converter/declaration.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    def find_city(self, registration_data):
+        pass
+
+    def find_spouse(self, relatives_data):
+        for relative_data in relatives_data:
+            relative = relative_data.get('subjectRelation')
+            if relative and relative in ['дружина', 'чоловік']:
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:32]: thank you), changed

File: business_register/converter/declaration.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                spouse_fullname = (
+                        relative_data['lastname']
+                        + ' ' + relative_data['firstname']
+                        + ' ' + relative_data['middlename']
+                ).lower()
+                return spouse_fullname
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:38]: changed

File: business_register/converter/declaration.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+        return None
+
+    def save_or_update_declaration(self):
+        for nacp_declarant_id in self.only_peps:
+            # getting general info including declaration id
+            response = requests.get(
+                settings.NACP_DECLARATION_LIST
+                + f'?user_declarant_id={nacp_declarant_id}'
+            )
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:47]: changed

File: business_register/converter/declaration.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:48]: removed

Comment (IegorSoboliev) [reply business_register/converter/declaration.py:48]: NACP returns various results)))

File: business_register/converter/declaration.py (hunk 2 above)
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:58]: We do need the declaration later in this Converter)

File: business_register/converter/declaration.py (hunk 3 above)
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:70]: changed. Consequences of learning JS)))

File: business_register/models/declaration_models.py (hunk 4 above)
Comment (IegorSoboliev) [reply business_register/models/declaration_models.py:41]: The reason is that one declarant can have many declarations

File: business_register/management/commands/fetch_peps_nacp_id.py (hunk 5 above)
Comment (IegorSoboliev) [reply business_register/management/commands/fetch_peps_nacp_id.py:60]: changed))

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**